tests/Doctrine/Tests/ORM/Functional/QueryTest.php

```diff
@@ -867,4 +867,40 @@ public function testMultipleJoinComponentsUsingLeftJoin()
         $this->assertInstanceOf(CmsUser::class, $users[2]);
         $this->assertNull($users[3]);
     }
+
+    public function testProcessParameterValueWithObjects()
+    {
+        $query = $this->_em->createQuery();
+
+        // non-mapped class
+        $dt = new \DateTime();
+        $this->assertSame(
+            $dt,
+            $query->processParameterValue($dt)
+        );
+
+        $user = new CmsUser();
+        $user->name = 'Thomas';
+        $user->username = 'fancyweb';
+
+        $this->_em->persist($user);
+        $this->_em->flush();
+
+        $this->assertEquals(true, $this->_em->getMetadataFactory()->hasMetadataFor(CmsUser::class));
+
+        // non-loaded metadata
+        $this->assertEquals(
+            $user->getId(),
+            $query->processParameterValue($user)
+        );
+
+        $this->_em->getMetadataFactory()->setMetadataFor(CmsUser::class, null);
```

Comment:
This is a bit of a hack... I can imagine that the test will break if we change something, unless this is a mock metadata factory?